### PR TITLE
Redis repos

### PR DESCRIPTION
**Pull request by nic-ball**
Completed the customer save and find behaviour tests, next will be writing the behat tests for gym owner and meals.
Changed the unique identifier to email and refactored the code accordingly.
Figured out that travis needs to know to install redis-server as a service...eventually.
Would be greatful if someone could take a look at the code and give me any pointers on where it could be improved or refactored down to more efficient :D
@mannion007 @braddle @stevestotter

### CODE REVIEW

File: src/Infrastructure/RedisCustomerRepository.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    public function findByEmail(Customer $email)
+    {
+        $this->client->connect('127.0.0.1', 6379);
+        $this->client->get($email);
+        $this->client->close();
+    }
+
+    public function signedUp(Customer $email)
```
Comment (mannion007) [inline src/Infrastructure/RedisCustomerRepository.php]: This function was not part of the interface we designed. It does the same thing as the save function - why do you need to add this?

File: composer.json [hunk 2]
```diff
@@ -6,12 +6,14 @@
...
+        "slim/slim": "^3.0",
+        "predis/predis": "^1.1"
     },
 
     "require-dev": {
         "phpspec/phpspec": "^3.0",
         "squizlabs/php_codesniffer": "2.7.*",
-        "behat/behat": "^3.3"
+        "behat/behat": "^3.3",
+        "phpunit/phpunit": "^6.1"
```
Comment (mannion007) [inline composer.json]: Doesn't look like you are using PHPUnit yet - I'd remove it for now. (You can use it if you want, although you could get by just fine with just PHP spec for this project)

File: .gitignore [hunk 3]
```diff
@@ -4,7 +4,8 @@
 
 #Dependencies
 
-
+dump.rdb
```
Comment (mannion007) [inline .gitignore:7]: What are these?

File: .gitignore [hunk 4]
```diff
@@ -4,7 +4,8 @@
...
 composer.phar
 composer.lock
 /vendor/
+test.php
```
Comment (mannion007) [inline .gitignore]: What are these?

File: features/redisCustomer.feature [hunk 5]
```diff
@@ -0,0 +1,27 @@
+Feature: Saving and finding a customer
```
Comment (mannion007) [inline features/redisCustomer.feature:1]: This feature file reads more like a unit test than a behavioural test. Remember that these scenarios are intended to be something you can discuss with the business owner and have them understand and contribute to.

Putting things in the database and getting them back out are purely technical concerns. The business owner will not understand or have any input into how this works, so there is no point in showing it to them.

Of course, this stuff does need to be tested, but Behat is not the correct tool to test your Redis Repository. You should instead use PHP Spec or PHP Unit to unit test the Redis Customer Repository.

**Review by mannion007 — CHANGES_REQUESTED** [review]
Good progress, I have added some feedback for you to look into and consider but I am impressed with what you have achieved.

File: src/Infrastructure/RedisCustomerRepository.php (hunk 1 above)
Comment (nic-ball) [reply src/Infrastructure/RedisCustomerRepository.php]: I have removed it. I got a little bit confused last week so am going to implement the feedback you've given cheers.

File: .gitignore (hunk 3 above)
Comment (nic-ball) [reply .gitignore:7]: I think this is something redis related, as it only appeared when I started using Redis

File: composer.json (hunk 2 above)
Comment (nic-ball) [reply composer.json]: Removed until it becomes necessary

File: .gitignore (hunk 3 above)
Comment (nic-ball) [reply .gitignore:7]: Yeah, it is for Redis so just ignoring it until i figure out exactly how it's used

File: .gitignore (hunk 4 above)
Comment (nic-ball) [reply .gitignore]: This was a test I made to check my redis functions, it has now been removed.

**Pull request merged**